File: Arrays and Hashing/347-Top-K-Frequent-Elements/347.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        std::unordered_map<int, int> count;
        std::vector<std::vector<int> > freq(nums.size() + 1);
        for(int n : nums){
            count[n] = 1 + count[n];
        }
        for(auto& it : count){
            freq[it.second].push_back(it.first);
        }
        std::vector<int> res;
        for(int i = freq.size() - 1; i > 0; i--){
            for(int num : freq[i]){
                res.push_back(num);
                if(res.size() == k){
                    return res;
                }
            }
        }
        return {};
    }
};

File: Arrays and Hashing/347-Top-K-Frequent-Elements/347.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number[]}
 */
var topKFrequent = function(nums, k) {
    const count = {};
    const freq = Array.from({length: nums.length + 1}, () => []);
    for(const n of nums){
        count[n] = 1 + (count[n] || 0);
    }
    for(const c in count){
        freq[count[c]].push(parseInt(c));
    }
    res = [];
    for(let i = freq.length - 1; i > 0; i--){
        for(const n of freq[i]){
            res.push(n);
            if(res.length == k){
                return res;
            }
        }
    }
};